pkg/controller/operators/catalog/operator.go

```diff
@@ -1191,21 +1237,42 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 						if len(matchedCSV) == 1 {
 							o.logger.Debugf("Found one owner for CRD %v", crd)
 
+							crdStatus := removeDeprecatedStoredVersions(currentCRD, &crd)
+							if crdStatus != nil {
+								currentCRD.Status.StoredVersions = crdStatus
+								resultCRD, err := o.opClient.ApiextensionsV1beta1Interface().ApiextensionsV1beta1().CustomResourceDefinitions().UpdateStatus(currentCRD)
+								if err != nil {
+									return errorwrap.Wrapf(err, "error updating CRD's status: %s", step.Resource.Name)
+								}
+								crd.SetResourceVersion(resultCRD.GetResourceVersion())
```

Comment:
I believe I misadvised in the past on this. Instead of pulling out the resource version, it's better to save off the result of the create here: https://github.com/operator-framework/operator-lifecycle-manager/pull/983/files#diff-efdf1c2d694e58ab3f44e9e02a4900b9R1224.

Reply:
@dinhxuanvu pointed out to me that the create has already failed at this point. Instead I propose doing a get on the CRD to get the proper fields filled in.